### PR TITLE
Port command p6 to new shell

### PR DESCRIPTION
**Pull request by PeiweiHu**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've documented or updated the documentation of every function and struct this PR changes. If not so I've explained why.
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**

+ port command p6[d/e] to new shell
+ add test for command p6[d/e]

**Test plan**

CI is green.

**Closing issues**

partially address https://github.com/rizinorg/rizin/issues/1590

### CODE REVIEW

File: librz/core/cmd/cmd_print.c [hunk 1]
```diff
@@ -7365,3 +7329,36 @@ RZ_IPI RzCmdStatus rz_cmd_disassemble_summarize_block_handler(RzCore *core, int
 	disasm_strings(core, input_cmd, NULL);
 	return RZ_CMD_STATUS_OK;
 }
+
+RZ_IPI RzCmdStatus rz_cmd_base64_encode_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+	ut32 malen = (core->blocksize * 4) + 1;
```
Comment (XVilka) [inline librz/core/cmd/cmd_print.c]: You can just inline it, it's only used once

File: librz/core/cmd/cmd_print.c [hunk 2]
```diff
@@ -7365,3 +7329,36 @@ RZ_IPI RzCmdStatus rz_cmd_disassemble_summarize_block_handler(RzCore *core, int
...
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_base64_encode((char *)buf, (const unsigned char *)core->block, core->blocksize);
+	rz_cons_println((const char *)buf);
+	rz_free(buf);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_cmd_base64_decode_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+	ut32 malen = (core->blocksize * 4) + 1;
```
Comment (XVilka) [inline librz/core/cmd/cmd_print.c]: Same

**Review by XVilka — CHANGES_REQUESTED** [review]

File: librz/core/cmd/cmd_print.c [hunk 3]
```diff
@@ -7365,3 +7329,33 @@ RZ_IPI RzCmdStatus rz_cmd_disassemble_summarize_block_handler(RzCore *core, int
 	disasm_strings(core, input_cmd, NULL);
 	return RZ_CMD_STATUS_OK;
 }
+
+RZ_IPI RzCmdStatus rz_cmd_base64_encode_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+	ut8 *buf = calloc(1, (core->blocksize * 4) + 1);
+
```
Comment (ret2libc) [inline librz/core/cmd/cmd_print.c]:
```suggestion
```

File: librz/core/cmd/cmd_print.c [hunk 4]
```diff
@@ -7365,3 +7329,33 @@ RZ_IPI RzCmdStatus rz_cmd_disassemble_summarize_block_handler(RzCore *core, int
...
+	}
+	rz_base64_encode((char *)buf, (const unsigned char *)core->block, core->blocksize);
+	rz_cons_println((const char *)buf);
+	rz_free(buf);
+	return RZ_CMD_STATUS_OK;
+}
+
+RZ_IPI RzCmdStatus rz_cmd_base64_decode_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+	ut8 *buf = calloc(1, (core->blocksize * 4) + 1);
+
```
Comment (ret2libc) [inline librz/core/cmd/cmd_print.c]:
```suggestion
```

File: librz/core/cmd/cmd_print.c [hunk 5]
```diff
@@ -7365,3 +7329,33 @@ RZ_IPI RzCmdStatus rz_cmd_disassemble_summarize_block_handler(RzCore *core, int
...
+RZ_IPI RzCmdStatus rz_cmd_base64_encode_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+	ut8 *buf = calloc(1, (core->blocksize * 4) + 1);
+
+	if (!buf) {
+		RZ_LOG_ERROR("Fail to allocate memory\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_base64_encode((char *)buf, (const unsigned char *)core->block, core->blocksize);
+	rz_cons_println((const char *)buf);
+	rz_free(buf);
```
Comment (ret2libc) [inline librz/core/cmd/cmd_print.c]: just use `free`. rz_free was introduced at some point but it was never really used... maybe worth removing at some point, but for now i don't think it really make sense given it's not used almost anywhere.

File: librz/core/cmd/cmd_print.c [hunk 6]
```diff
@@ -7365,3 +7329,33 @@ RZ_IPI RzCmdStatus rz_cmd_disassemble_summarize_block_handler(RzCore *core, int
...
+		RZ_LOG_ERROR("Fail to allocate memory\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	if (rz_base64_decode(buf, (const char *)core->block, core->blocksize) < 0) {
+		RZ_LOG_ERROR("rz_base64_decode: invalid stream\n");
+		rz_free(buf);
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_cons_println((const char *)buf);
+	rz_free(buf);
```
Comment (ret2libc) [inline librz/core/cmd/cmd_print.c]:
```suggestion
	free(buf);
```

File: librz/core/cmd/cmd_print.c [hunk 7]
```diff
@@ -7365,3 +7329,33 @@ RZ_IPI RzCmdStatus rz_cmd_disassemble_summarize_block_handler(RzCore *core, int
...
 }
+
+RZ_IPI RzCmdStatus rz_cmd_base64_encode_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+	ut8 *buf = calloc(1, (core->blocksize * 4) + 1);
+
+	if (!buf) {
+		RZ_LOG_ERROR("Fail to allocate memory\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	rz_base64_encode((char *)buf, (const unsigned char *)core->block, core->blocksize);
```
Comment (ret2libc) [inline librz/core/cmd/cmd_print.c]: use `rz_base64_encode_dyn` without the calloc.

File: librz/core/cmd/cmd_print.c [hunk 8]
```diff
@@ -7365,3 +7329,33 @@ RZ_IPI RzCmdStatus rz_cmd_disassemble_summarize_block_handler(RzCore *core, int
...
+}
+
+RZ_IPI RzCmdStatus rz_cmd_base64_decode_handler(RzCore *core, int argc, const char **argv, RzOutputMode mode) {
+	ut8 *buf = calloc(1, (core->blocksize * 4) + 1);
+
+	if (!buf) {
+		RZ_LOG_ERROR("Fail to allocate memory\n");
+		return RZ_CMD_STATUS_ERROR;
+	}
+	if (rz_base64_decode(buf, (const char *)core->block, core->blocksize) < 0) {
```
Comment (ret2libc) [inline librz/core/cmd/cmd_print.c]: use `rz_base64_decode_dyn`

**Review by ret2libc — CHANGES_REQUESTED** [review]

**Comment by PeiweiHu** [comment]
@ret2libc Thanks! The code is neater.

**Review by ret2libc — APPROVED** [review]

**Review by XVilka — APPROVED** [review]

**Pull request merged**